cpp/example/src/process_csv.cpp

```diff
@@ -0,0 +1,92 @@
+#include <iostream>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+#include <rmm/mr/device/per_device_resource.hpp>
+#include <rmm/mr/device/pool_memory_resource.hpp>
+
+#include <cudf/aggregation.hpp>
+#include <cudf/groupby.hpp>
+#include <cudf/table/table.hpp>
+
+#include <cudf/io/csv.hpp>
+
+std::unique_ptr<cudf::table> read_csv(std::string const& file_path)
+{
+  auto source_info    = cudf::io::source_info(file_path);
+  auto builder        = cudf::io::csv_reader_options::builder(source_info);
+  auto options        = builder.build();
+  auto data_with_meta = cudf::io::read_csv(options);
+  auto data           = std::move(data_with_meta.tbl);
+  return data;
+}
+
+void write_csv(cudf::table_view const& tbl_view, std::string const& file_path)
+{
+  auto sink_info = cudf::io::sink_info(file_path);
+  auto builder   = cudf::io::csv_writer_options::builder(sink_info, tbl_view);
+  auto options   = builder.build();
+  cudf::io::write_csv(options);
+}
+
+std::unique_ptr<cudf::table> get_columns_from_table(cudf::table_view table,
+                                                    std::vector<int> const& indices)
+{
+  std::vector<std::unique_ptr<cudf::column>> cols;
+  for (auto i : indices) {
+    cols.push_back(std::move(std::make_unique<cudf::column>(table.column(i))));
+  }
```

Comment:
could use a std::transform here.

Reply:
I realized in this example (getting two columns from a table), `get_column_from_table` seems excessive. Removed.